include/solvers/heat_transfer.h

```diff
@@ -0,0 +1,273 @@
+   */
+  virtual void
+  attach_solution_to_output(DataOut<dim> &data_out) override;
+
+
+  /**
+   * @brief Calculates the L2 error of the solution
+   */
+  double
+  calculate_L2_error();
+
+
+  /**
+   * @brief Carry out the operations required to finish a simulation correctly.
+   */
+  virtual void
+  finish_simulation() override;
+
+  /**
+   * @brief Carry out the operations require to finish a time step correctly. This
+   * includes setting the previous values
+   */
+  virtual void
+  finish_time_step() override;
+
+  /**
+   * @brief Rearrange vector solution correctly for transient simulations
+   */
+  virtual void
+  percolate_time_vectors() override;
+
+  /**
+   * @brief Postprocess the auxiliary physics results. Post-processing this case implies
+   * the calculation of all derived quantities using the solution vector of the
+   * physics. It does not concern the output of the solution using the
+   * DataOutObject, which is accomplished through the attach_solution_to_output
+   * function
+   */
+  virtual void
+  postprocess(bool first_iteration) override;
+
+
+  /**
+   * @brief pre_mesh_adaption Prepares the auxiliary physics variables for a
+   * mesh refinement/coarsening
+   */
+  virtual void
+  pre_mesh_adaptation();
+
+  /**
+   * @brief post_mesh_adaption Interpolates the auxiliary physics variables to the new mesh
+   */
+  virtual void
+  post_mesh_adaptation();
+
+
+  /**
+   * @brief Returns the dof_handler of the heat transfer physics
+   */
+  virtual const DoFHandler<dim> &
+  get_dof_handler() override
+  {
+    return dof_handler;
+  };
+
+  /**
+   * @brief Sets-up the DofHandler and the degree of freedom associated with the physics.
+   */
+  virtual void
+  setup_dofs() override;
+
+  /**
+   * @brief Sets-up the initial conditions associated with the physics. Generally, physics
+   * only support imposing nodal values, but some physics additionnaly support
+   * the use of L2 projection or steady-state solutions.
+   */
+  virtual void
+  set_initial_conditions() override;
+
+  /**
+   * @brief Call for the solution of the linear system of equation using a strategy appropriate
+   * to the auxiliary physics
+   *
+   * @param initial_step Provides the linear solver with indication if this solution is the first
+   * one for the system of equation or not
+   *
+   * @param renewed_matrix Indicates to the linear solve if the system matrix has been recalculated or not
+   */
+  virtual void
+  solve_linear_system(const bool initial_step,
+                      const bool renewed_matrix = true);
+
+
+  /**
+   * @brief Getter methods to get the private attributes for the physic currently solved
+   *
+   * @param number_physic_current Indicates the number associated with the physic currently solved
+   * default value = 0, meaning only one physic, generally associated with the
+   * flow equations, is solved by default
+   */
+  virtual TrilinosWrappers::MPI::Vector &
+  get_evaluation_point() override
```

Comment:
The getters are member of auxiliary physics (otherwise they could not be overriden). In auxiliary physics they are templated by the VectorType, here VectorType is specialized to TrlinosWrappers::MPI::Vector and hence it overwrites the mother function. It's a weird Polymorphism/Template hybrid shenanigan.

Reply:
Ok I did not checked that these getters were in PhysicsSolver, I get it now :)